### DISCUSSION TITLE
Simulation Results

### DISCUSSION THREAD

**Discussion by wdmhouston** [GEOS-DEV/GEOS#1428] [Q&A] [answered]
Hi experts,

    I'm testing GEOSX and trying to visualize the simulation results from the CO2 injection example. I have several questions:

1. In paraview, which property can be used to identify CO2 distribution? 
2. How to request pressure of selected cells and write to HD5? I need to make a plot.
3. How to check total CO2(mass and volume) injected into reservoir?
4. How to request results of the following: CO2 dissolved in water phase, trapped in gas phase and in mobile phase.

Thanks and Regards,

wdmhouston

**Comment by francoishamon** [comment, marked as answer]
Hello,

1. In Paraview, you can visualize the CO2 saturation field (fraction of pore volume occupied by the CO2 phase) by clicking `phaseVolumeFraction` in the cell-based properties that you can display on the mesh. Then, by default, Paraview will show a `phaseVolumeFraction` `Magnitude` that does not make much physical sense, so you may want to unclick `Magnitude` and select instead `X` for the saturation of the CO2 phase and `Y` for the saturation of brine.
2. To request the pressure of selected cells and write to HDF5 you can do the following:

2.a. In the `Geometry` block of the XML, you define a `Box` that selects a few cells in the mesh by their positions. The selected cells are those that are fully contained in the `Box` volume:
```
<Box
      name="theLocationWhereIWantToPlotPressure"
      xMin="{-1, -1, -1}"
      xMax="{1.1, 1.1, 1.1}"/>
```
2.b. In the `Events` block of the XML, you define two `PeriodicEvent`s. One is the `PeriodicEvent` that defines the frequency at which you collect the pressure values:
```
<PeriodicEvent
      name="timeHistoryCollection"
      timeFrequency="5e5"
      targetExactTimestep="1"
      target="/Tasks/pressureCollection" />
```
and the other is the `PeriodicEvent` that defines the frequency at which you output the values:
```
 <PeriodicEvent
      name="timeHistoryOutput"
      timeFrequency="1.5e7"
      targetExactTimestep="1"
      target="/Outputs/timeHistoryOutput" />
```
2.c. In the `Outputs` block of the XML, you define the `TimeHistory` output that is going to write to file:
```
<TimeHistory
      name="timeHistoryOutput"
      sources="{/Tasks/pressureCollection}"
      filename="pressureHistory" />
```
2.d. In the `Task` block, you define the `PackCollection` task that is going to collect the pressure values in the `Box` that you have defined:
```
    <PackCollection
      name="pressureCollection"
      objectPath="ElementRegions/nameOfTheRegion"
      setNames="{theLocationWhereIWantToPlotPressure}"
      fieldName="pressure" />
```
Make sure you include the `setNames` that will allow GEOSX to only plot the cells that you have selected. If something does not work, please post your xml file in an issue and we will do the the debugging together.

3. You can use the same mechanism as above, following the example [here](https://github.com/GEOSX/GEOSX/blob/develop/src/coreComponents/physicsSolvers/fluidFlow/benchmarks/Egg/dead_oil_egg.xml). Currently this will output the mass rates for all the well cells (which is probably more that you want), but we are working on a modified `TimeHistory` that will allow us to plot just one rate (mass and volume) per well.
4. The quantity to look for if `phaseCompFraction`, that shows the amount of CO2 component dissolved into the water phase. Unfortunately, we deactivated the output of this quantity in Paraview because of some technical issues in the VTK writer. So, it is currently unavailable for plotting in Paraview. Let me post an issue, and we will bring it back.

**Reply by wdmhouston** [reply]
Francois,
    Thank you for your prompt response to my questions. 
    For question 3, it is still a big challenge to calculate rate or total injection using cell data for each well(especially when there are multiple wells involved) .  It will be great to let end user request both well injection/production rate and injection/production total over time.
    For question 4, can we add CO2 dissolved/mobile/trapped to TimeHistory output? These outputs are critical for CO2 case.

Thanks and Regards,

wdmhouston

**Reply by AbrightWay** [reply]
Dear @francoishamon ,

I'm quite confused between timeHistoryCollection and timeHistoryOutput:  as far as I understand, timeHistoryCollection collects pressure at 5e5 frequency and saves it temporarily, and then, timeHistoryOutput queries the saved pressure at timeHistoryCollection's timesteps and writes it to HDF5 file at 1.5e7 frequency? Does this mean the simulation have to run extra time-steps if the timeFrequency of the solver is bigger (different) than the timeFrequency of timeHistoryCollection?

If my intuition about timeHistoryCollection and timeHistoryOutput, why don't we just use the same timeFrequency for both or even just use either one of them?

Best,
Trinh
